include/eigensolve_quda.h

```diff
@@ -273,7 +276,7 @@ namespace quda
        @brief Compute eigenvalues and their residiua.  This variant compute the number of converged eigenvalues.
        @param[in] mat Matrix operator
        @param[in] evecs The eigenvectors
-       @param[in] evals The eigenvalues
+       @param[in,out] evals The eigenvalues
```

Comment:
Now that we have a 3-d eigensolver, we should be make the `computeEvals` routine virtual and in the case of the 3-d eigensolvers make a custom override that `errorQuda`s out. This applies to many other routines as well---essentially anything that has a `3D` version in `TRLM3D`. We may also want to create a virtual function `virtual bool is_3d_eigensolver()` that by default returns `false` and for the 3-d eigensolvers returns `true`.

Reply:
[1e2c0c7](https://github.com/lattice/quda/pull/1506/commits/1e2c0c7cf2da19395b92e18dcf4996a38087afd0) does something related: remove code replication and make `computeEvals3D` a specialization of the regular `computeEvals` which is what I was planning on doing anyway.

Marking as resolved, but we can discuss this more if needed.